src/frontends/lean/pp.cpp

```diff
@@ -2103,9 +2103,9 @@ formatter_factory mk_pretty_formatter_factory() {
                 std::function<bool(sexpr const &, sexpr const &)> sep; // NOLINT
                 token_table const * last = nullptr;
                 sep = [&](sexpr const & s1, sexpr const & s2) {
-                    bool b;
-                    mk_pair(b, last) = fn_ptr->needs_space_sep(last, sexpr_to_string(s1), sexpr_to_string(s2));
```

Comment:
It's not surprising that this has no effect.  The `mk_pair` function takes two `const &` references, and hence won't modify them.  What happens instead is that a temporary `pair` object is created with *copies* of `b` and `last`, and the `first` and `second` fields of this temporary object are assigned.